src/core/interaction/trackball.cpp

```diff
@@ -833,6 +1022,22 @@ void Trackball::panDown(Event* event) {
     event->markAsUsed();
 }
 
+/* \brief Zooms on mouse wheel event
+ *
+ * Uses the step zoom functions and triggers one
+ * step per mouse wheel tick. Horizontal scrolling is disregarded
+ */
+void Trackball::zoomWheel(Event* event) {
+    auto wheelEvent = static_cast<WheelEvent*>(event);
+    int steps = static_cast<int>(wheelEvent->delta().y);
+
+    for (size_t i = 0; i < abs(steps); ++i)
+        if (steps > 0)
+            zoomIn(event);
```

Comment:
This will cause one property invalidation per iteration. Better to refactor zoomIn/stepZoom to take number of steps=